modules/src/timestamp.rs

```diff
@@ -88,42 +90,48 @@ impl Timestamp {
     /// than the current or if either of the `Timestamp`s is not set.
     pub fn duration_since(&self, other: &Timestamp) -> Option<Duration> {
         match (self.time, other.time) {
-            (Some(time1), Some(time2)) => time1.signed_duration_since(time2).to_std().ok(),
+            (Some(time1), Some(time2)) => time1.duration_since(time2).ok(),
             _ => None,
         }
     }
 
-    /// Convert a `Timestamp` from [`chrono::DateTime<Utc>`].
-    pub fn from_datetime(time: DateTime<Utc>) -> Timestamp {
-        Timestamp { time: Some(time) }
+    /// Convert a `Timestamp` to `u64` value in nanoseconds. If no timestamp
+    /// is set, the result is 0.
+    ///
+    #[deprecated(since = "0.9.1", note = "use `nanoseconds` instead")]
+    pub fn as_nanoseconds(&self) -> u64 {
+        (*self).nanoseconds()
     }
 
     /// Convert a `Timestamp` to `u64` value in nanoseconds. If no timestamp
     /// is set, the result is 0.
     /// ```
     /// use ibc::timestamp::Timestamp;
+    ///
     /// let max = u64::MAX;
     /// let tx = Timestamp::from_nanoseconds(max).unwrap();
-    /// let utx = tx.as_nanoseconds();
+    /// let utx = tx.nanoseconds();
     /// assert_eq!(utx, max);
     /// let min = u64::MIN;
     /// let ti = Timestamp::from_nanoseconds(min).unwrap();
-    /// let uti = ti.as_nanoseconds();
+    /// let uti = ti.nanoseconds();
     /// assert_eq!(uti, min);
+    /// let tz = Timestamp::default();
+    /// let utz = tz.nanoseconds();
+    /// assert_eq!(utz, 0);
     /// ```
-    pub fn as_nanoseconds(&self) -> u64 {
+    pub fn nanoseconds(self) -> u64 {
```

Comment:
Why does this consume `self`?

Reply:
`Self` is `Copy`, so it does not matter much.

Reply:
If there's a concern about performance, I'd rather put the `#[inline]` attribute on the method so the compiler can always optimize it in the best way.